insights/parsers/nfnetlink_queue.py

```diff
@@ -1,67 +1,67 @@
-from insights.specs import nfnetlink_queue
+"""
+NfnetLinkQueue - file ``/proc/net/netfilter/nfnetlink_queue``
+=============================================================
 
+Reads the ``/proc/net/netfilter/nfnetlink_queue`` file and creates a list of
+dictionaries, one dictionary per row in the file.
 
-@parser(nfnetlink_queue)
-class NfnetLinkQueue(Parser):
-    """Reads the ``/proc/net/netfilter/nfnetlink_queue`` file and
-    creates a list of dictionaries, one dictionary per row in the file.
+The keys of the dictionary are (see
+https://home.regit.org/netfilter-en/using-nfqueue-and-libnetfilter_queue/):
+
+- ``queue_number``
+- ``peer_portid``: good chance it is process ID of software listening to
+  the queue
+- ``queue_total``: current number of packets waiting in the queue
+- ``copy_mode``: 0 and 1 only message only provide meta data. If 2, the
+  message provides a part of packet of size copy range.
+- ``copy_range``: length of packet data to put in message
+- ``queue_dropped``: number of packets dropped because queue was full
+- ``user_dropped``: number of packets dropped because netlink message
+  could not be sent to userspace. If this counter is not zero, try
+  to increase netlink buffer size. On the application side, you will
+  see gap in packet id if netlink message are lost.
+- ``id_sequence``: packet id of last packet
+- The last field is always '1' and is ignored.
+
+Example Input::
 
-    The keys of the dictionary are (see
-    https://home.regit.org/netfilter-en/using-nfqueue-and-libnetfilter_queue/):
-    - queue_number
-    - peer_portid: good chance it is process ID of software listening to
-      the queue
-    - queue_total: current number of packets waiting in the queue
-    - copy_mode: 0 and 1 only message only provide meta data. If 2
-      message provide a part of packet of size copy range.
-    - copy_range: length of packet data to put in message
-    - queue_dropped: number of packets dropped because queue was full
-    - user_dropped: number of packets dropped because netlink message
-      could not be sent to userspace. If this counter is not zero, try
-      to increase netlink buffer size. On the application side, you will
-      see gap in packet id if netlink message are lost.
-    - id_sequence: packet id of last packet
+    0  -4423     0 2 65535     0     0       22  1
+    1  -4424     0 2 65535     0     0       27  1
 
-    Example Input:
+Examples:
 
-        0  -4423     0 2 65535     0     0       22  1
-        1  -4424     0 2 65535     0     0       27  1
+    >>> # Set up of the environment - ignore this bit:
+    >>> nfnetlink_queue_data = '''
+    ... 0  -4423     0 2 65535     0     0       22  1
+    ... 1  -4424     0 2 65535     0     0       27  1
+    ... '''
+    >>> from insights.tests import context_wrap
+    >>> from insights.parsers.nfnetlink_queue import NfnetLinkQueue
+    >>> shared = {NfnetLinkQueue: NfnetLinkQueue(context_wrap(nfnetlink_queue_data))}
```

Comment:
@PaulWay since `shared` is no longer used by consumers of parsers in versions after 1.x, this could be simplified to be the suggested name that would be used in a rule, for example instead of `nf` you might have:
```
nfnetlink_queue = NfnetLinkQueue(context_wrap(nfnetlink_queue_data))
```

Reply:
Good point!  I'll have to watch out for that in the future.  Thanks for picking that up @bfahr